### PR TITLE
Add dateFieldsFreeInline tag for freetext date entry

### PR DESCRIPTION
**Pull request by gianick**

### CODE REVIEW

File: src/main/twirl/uk/gov/hmrc/play/views/helpers/dateFieldsFreeInline.scala.html [hunk 1]
```diff
@@ -0,0 +1,58 @@
+@*
+* Copyright 2015 HM Revenue & Customs
```
Comment (callumcodes) [inline src/main/twirl/uk/gov/hmrc/play/views/helpers/dateFieldsFreeInline.scala.html]: 2016?

Comment (gianick) [inline src/main/twirl/uk/gov/hmrc/play/views/helpers/dateFieldsFreeInline.scala.html]: Fixed

File: src/main/twirl/uk/gov/hmrc/play/views/helpers/dateFieldsFreeInline.scala.html [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+*@
+@(formItem:Form[_], fieldName: String, args: (Symbol,Any)*)(implicit lang: play.api.i18n.Lang)
+
+@import uk.gov.hmrc.play.mappers.DateFields._
+@import play.api.i18n._
+@import uk.gov.hmrc.play.mappers.DateFormatSymbols._
+
+
+
+@fielsetClasses = @{if(formItem(fieldName).hasErrors) "form-field--error" else "" }
```
Comment (benaveryee) [inline src/main/twirl/uk/gov/hmrc/play/views/helpers/dateFieldsFreeInline.scala.html]: typo: should this be fieldsetClasses?

Comment (gianick) [inline src/main/twirl/uk/gov/hmrc/play/views/helpers/dateFieldsFreeInline.scala.html]: Fixed now thanks

File: src/main/twirl/uk/gov/hmrc/play/views/helpers/dateFieldsFreeInline.scala.html [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+@fieldsetClasses = @{if(formItem(fieldName).hasErrors) "form-field--error" else "" }
+
+<fieldset class="form-group form-date @fieldsetClasses" id="@fieldName-fieldset">
+
+    @formItem(fieldName).errors.map { error => <span class="error-notification">@Messages(error.message)</span>}
+
+    @input(
+        formItem(s"${fieldName}.day"),
+        '_label -> Messages("date.fields.day"),
+        '_labelClass -> "form-group form-group-day label--inlineTextinput form-field--inline-text-input",
```
Comment (rpowis) [inline src/main/twirl/uk/gov/hmrc/play/views/helpers/dateFieldsFreeInline.scala.html]: `label--inlineTextinput` doesn't exist in assets-frontend. Can you removed this please?

Comment (rpowis) [inline src/main/twirl/uk/gov/hmrc/play/views/helpers/dateFieldsFreeInline.scala.html]: `form-field--inline-text-input` also isn't needed for outputting this. It'll also avoid the need for `margin: 0` in https://github.com/hmrc/assets-frontend/pull/534

File: src/main/scala/uk/gov/hmrc/play/mappers/DateTuple.scala [hunk 4]
```diff
@@ -77,6 +78,23 @@ trait DateTuple {
   }
   )
 
+  def validDateTuple: Mapping[DateTime] = {
+
+    def verifyDigits(triple: (String, String, String )) =
+      !triple._1.exists(!_.isDigit) && !triple._2.exists(!_.isDigit) && !triple._3.exists(!_.isDigit)
```
Comment (anshulbajpai) [inline src/main/scala/uk/gov/hmrc/play/mappers/DateTuple.scala:84]: Use forAll to simplify

**Comment by anshulbajpai** [comment]
Add tests

**Pull request closed without merging**